File: 5th course/java se (basics)/Task 5 [Math] Triangle/src/by/gsu/lab/beans/Triangle.java
package by.gsu.lab.beans;

public class Triangle {

	private final double firstSide;
	private final double secondSide;
	private final double thirdSide;

	public Triangle(double firstSide, double secondSide, double thirdSide) {
		if (!isTriangleExists(firstSide, secondSide, thirdSide)) {
			throw new IllegalArgumentException();
		}
		this.firstSide = firstSide;
		this.secondSide = secondSide;
		this.thirdSide = thirdSide;
	}

	/**
	 * Verifies the existence of a triangle.
	 */
	public static boolean isTriangleExists(double firstSide, double secondSide,
			double thirdSide) {
		return (0 > Double.compare(thirdSide, (firstSide + secondSide)))
			&& (0 > Double.compare(secondSide, (firstSide + thirdSide)))
			&& (0 > Double.compare(firstSide, (secondSide + thirdSide)));
	}

	public double getPerimeter() {
		return firstSide + secondSide + thirdSide;
	}

	/**
	 * Getting the area of a triangle
	 */
	public double getArea() {
		double semiPerimeter = getPerimeter() / 2;
		double buffer = semiPerimeter * (semiPerimeter - firstSide)
			* (semiPerimeter - secondSide) * (semiPerimeter - thirdSide);
		return Math.sqrt(buffer);
	}

	// getters

	public double getFirstSide() {
		return firstSide;
	}

	public double getSecondSide() {
		return secondSide;
	}

	public double getThirdSide() {
		return thirdSide;
	}

	@Override
	public String toString() {
		final String DELIMETER = ", ";
		return "Triangle: " + firstSide + DELIMETER + secondSide 
			+ DELIMETER + thirdSide;
	}
}

File: others/labs (c_sharp, wpf, unit testing)/Task5/Task5/Triangle.cs
using System;

namespace Task5
{
    /// <summary>
    /// Треугольник
    /// </summary>
    public class Triangle
    {
        private double _firstSide;
        private double _secondSide;
        private double _thirdSide;

        public Triangle(double firstSide, double secondSide, double thirdSide)
        {
            if (!isTriangleExists(firstSide, secondSide, thirdSide))
            {
                throw new ArgumentException("Треугольник с такими сторонами не существует");
            }
            _firstSide = firstSide;
            _secondSide = secondSide;
            _thirdSide = thirdSide;
        }

        /// <summary>
        /// Проверка существования треугольника
        /// </summary>
        public static bool isTriangleExists(double firstSide, double secondSide,
                double thirdSide)
        {
            return (0 > thirdSide.CompareTo(firstSide + secondSide)) &&
                (0 > secondSide.CompareTo(firstSide + thirdSide)) &&
                (0 > firstSide.CompareTo(secondSide + thirdSide));
        }

        /// <summary>
        /// Рассчёт периметра треугольника
        /// </summary>
        public double Perimeter()
        {
            return _firstSide + _secondSide + _thirdSide;
        }

        /// <summary>
        /// Рассчёт площади треугольника
        /// </summary>
        public double Area()
        {
            double semiPerimeter = Perimeter() / 2;
            double buffer = semiPerimeter * (semiPerimeter - _firstSide)
                                          * (semiPerimeter - _secondSide)
                                          * (semiPerimeter - _thirdSide);
            return Math.Sqrt(buffer);
        }

        public double FirstSide
        {
            get
            {
                return _firstSide;
            }
        }

        public double SecondSide
        {
            get
            {
                return _secondSide;
            }
        }

        public double ThirdSide
        {
            get
            {
                return _thirdSide;
            }
        }
    }
}
